src/benchmark/tpcds_benchmark.cpp

```diff
@@ -0,0 +1,142 @@
+  // clang-format on
+
+  std::shared_ptr<opossum::BenchmarkConfig> config;
+  int scale_factor;
+
+  if (opossum::CLIConfigParser::cli_has_json_config(argc, argv)) {
+    // JSON config file was passed in
+    const auto json_config = opossum::CLIConfigParser::parse_json_config_file(argv[1]);
+    scale_factor = json_config.value("scale", 1);
+    config = std::make_shared<opossum::BenchmarkConfig>(
+        opossum::CLIConfigParser::parse_basic_options_json_config(json_config));
+
+  } else {
+    // Parse regular command line args
+    const auto cli_parse_result = cli_options.parse(argc, argv);
+
+    if (CLIConfigParser::print_help_if_requested(cli_options, cli_parse_result)) {
+      return 0;
+    }
+    scale_factor = cli_parse_result["scale"].as<int>();
+
+    config =
+        std::make_shared<opossum::BenchmarkConfig>(opossum::CLIConfigParser::parse_basic_cli_options(cli_parse_result));
+  }
+
+  // For the data generation, the official tpc-ds toolkit is used.
+  // This toolkit does not provide the option to generate data using a scale factor less than 1.
+  Assert(scale_factor >= 1, "For now, TPC-DS benchmark only supports scale factor 1");
+
+  auto context = opossum::BenchmarkRunner::create_context(*config);
+
+  std::cout << "- TPC-DS scale factor is " << scale_factor << std::endl;
+
+  // TPC-DS FileBasedQueryGenerator specification
+  std::optional<std::unordered_set<std::string>> query_subset;
+  const auto query_filename_blacklist = std::unordered_set<std::string>{};
+  std::string query_path = "resources/benchmark/tpcds/queries/supported";
+  std::string table_path = "resources/benchmark/tpcds/tables";
+
+  Assert(std::filesystem::is_directory(query_path), "Query path (" + query_path + ") has to be a directory.");
+  Assert(std::filesystem::is_directory(table_path), "Table path (" + table_path + ") has to be a directory.");
+  Assert(std::filesystem::exists(std::filesystem::path{query_path + "/query_07.sql"}), "Queries have to be available.");
+  Assert(std::filesystem::exists(std::filesystem::path{table_path + "/call_center.csv.json"}),
+         "Table schemes have to be available.");
+
+  if (!data_files_available(table_path)) {
+    if (std::filesystem::exists(std::filesystem::path{"third_party/tpcds-dsdgen/dsdgen"})) {
+      const auto setup_table_data_files_return_code = system(
+          "cd ./third_party/tpcds-dsdgen/ &&"
+          "./dsdgen -scale 1 -dir ../../resources/benchmark/tpcds/tables -terminate n -verbose -f &&"
+          "cd ../../resources/benchmark/tpcds/tables &&"
+          "for x in *.dat; do mv $x ${x%.dat}.csv; done &&"
+          "cd ../../../../");
+      Assert(setup_table_data_files_return_code == 0, "Generating table data files failed.");
+    } else {
+      Fail("Could not find 'third_party/tpcds-dsdgen/dsdgen'. Did you run the benchmark from the project root dir?");
+    }
+  }
+
+  Assert(data_files_available(table_path), "Table data files are missing.");
```

Comment:
“Missing” sounds like they should have been there. In this case, they failed to be generated